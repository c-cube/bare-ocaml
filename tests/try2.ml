module E = Example1
module Bare = Bare_encoding

let () =
  print_endline ">>> run encoding test 2";
  let p1 =
    E.Person.Employee
      {
        E.Employee.name = "bar";
        email = "bar@iHaveAVeryLongEmailRight.edu.ac.co.uk";
        department = E.Department.CUSTOMER_SERVICE;
        hireDate = "2020-04-01";
        publicKey = Some (Bytes.make 128 'A');
        metadata =
          Bare.String_map.singleton "level" (Bytes.of_string "triple A grade!");
        address =
          {
            E.Address.address = [| "12"; "business avenue"; "front"; "center" |];
            city = "Business City";
            state = "HQ";
            country = "Office Space 1";
          };
      }
  in
  let s = Bare.to_string E.Person.encode p1 in
  (let oc = open_out "bar.data" in
   output_string oc s;
   flush oc;
   close_out oc);
  let p2 = Bare.of_string_exn E.Person.decode s in
  let s2 = Bare.to_string E.Person.encode p2 in
  (let oc = open_out "bar2.data" in
   output_string oc s2;
   flush oc;
   close_out oc);
  assert (s = s2);
  assert (p1 = p2);
  ()
