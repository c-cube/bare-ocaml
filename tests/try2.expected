>>> run encoding test 2
p1:
00000000: 0103 6261 7229 6261 7240 6948 6176 6541  ..bar)bar@iHaveA
00000001: 5665 7279 4c6f 6e67 456d 6169 6c52 6967  VeryLongEmailRig
00000002: 6874 2e65 6475 2e61 632e 636f 2e75 6b04  ht.edu.ac.co.uk.
00000003: 0231 320f 6275 7369 6e65 7373 2061 7665  .12.business ave
00000004: 6e75 6505 6672 6f6e 7406 6365 6e74 6572  nue.front.center
00000005: 010d 4275 7369 6e65 7373 2043 6974 7901  ..Business City.
00000006: 0248 5101 0e4f 6666 6963 6520 5370 6163  .HQ..Office Spac
00000007: 6520 3102 0a32 3032 302d 3034 2d30 3101  e 1..2020-04-01.
00000008: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000009: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000010: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000011: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000012: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000013: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000014: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000015: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000016: 0105 6c65 7665 6c0f 7472 6970 6c65 2041  ..level.triple A
00000017: 2067 7261 6465 21                         grade!

p2:
00000000: 0103 6261 7229 6261 7240 6948 6176 6541  ..bar)bar@iHaveA
00000001: 5665 7279 4c6f 6e67 456d 6169 6c52 6967  VeryLongEmailRig
00000002: 6874 2e65 6475 2e61 632e 636f 2e75 6b04  ht.edu.ac.co.uk.
00000003: 0231 320f 6275 7369 6e65 7373 2061 7665  .12.business ave
00000004: 6e75 6505 6672 6f6e 7406 6365 6e74 6572  nue.front.center
00000005: 010d 4275 7369 6e65 7373 2043 6974 7901  ..Business City.
00000006: 0248 5101 0e4f 6666 6963 6520 5370 6163  .HQ..Office Spac
00000007: 6520 3102 0a32 3032 302d 3034 2d30 3101  e 1..2020-04-01.
00000008: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000009: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000010: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000011: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000012: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000013: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000014: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000015: 4141 4141 4141 4141 4141 4141 4141 4141  AAAAAAAAAAAAAAAA
00000016: 0105 6c65 7665 6c0f 7472 6970 6c65 2041  ..level.triple A
00000017: 2067 7261 6465 21                         grade!

